docs/unit_tests.md

```diff
@@ -0,0 +1,98 @@
+1. A TestRule class that is essentially and atlas or multiple atlas to test against.
+2. A `ConsumerBasedExpectedCheckVerifier` instance that will execute the check you are testing against the test atlas.
+3. A series of units test using the JUnit framework.
+
+### The `ConsumerBasedExpectedCheckVerifier`
+
+The ConsumerBasedExpectedCheckVerifier is a class that will run your check against a test atlas. The test atlas are often very small atlas objects that contain specific data to test against. This verifier has too main types of functions
+
+1. `actual(Atlas, BaseCheck)` - The actual method establishes what data to run against (ie. what atlas file) and what check to run against the provided test atlas.
+2. verify functions - The verify functions will be used to verify the results of the running the check over the test atlas file. Below are some useful functions
+    
+    - `verifyNotEmpty()` - Verifies that at least 1 flag was produced by the check
+    - `verifyEmpty()` - Verifies that no flags were produced by the check
+    - `verifyExpectedSize(int)` - Verifies that a specific number of flags were produced by the check
+    - `verify(Consumer<List<CheckFlag>>)` - A custom verifier allowing the developer to loop through each individual flag that is produced and verify it in any way that the developer wishes.
+    
+ ### The `TestRule`
+ 
+ As mentioned previously the TestRule is a class that is associated with the Unit test class and contains all the test data, which would essentially be an Atlas in some form or another. An Atlas can be created in 2 primary ways:
+ 
+ - Inline - An inline atlas uses tags to build it within the code. The code below will create an inline atlas that contains 3 nodes and 2 edges. You can also use the inline @tags to build an Atlas containing `lines`, `points` or `relations`. An example of this can be found in the [FloatingEdgeCheckTestRule](../src/test/java/org/openstreetmap/atlas/checks/validation/linear/edges/FloatingEdgeCheckTestRule.java)
+ ```java
+@TestAtlas(nodes = { @Node(coordinates = @Loc(value = "37.32544,-122.033948")),
+            @Node(coordinates = @Loc(value = "37.33531,-122.009566")),
+            @Node(coordinates = @Loc(value = "37.3314171,-122.0304871")) }, edges = {
+                    @Edge(coordinates = { @Loc(value = "37.32544,-122.033948"), @Loc(value = "37.33531,-122.009566") }, tags = {
+                            "highway=SECONDARY" }),
+                    @Edge(coordinates = { @Loc(value = "37.33531,-122.009566"), @Loc(value = "37.3314171,-122.0304871") }, tags = {
+                            "highway=SECONDARY" }) })
+```
+- Atlas Text Files - It is fairly easy to produce very small test atlas text files which that can be stored in resources much like the [poolsize.altas](../src/test/resources/org/openstreetmap/atlas/checks/validation/areas/poolsize.atlas). And generating this files are fairly straight forward, and can be done by following these steps:
```

Comment:
Small note - we usually avoid zipping these up, and store in raw .txt format so it's easier to see what's in the atlas file with the naked eye. Might be worth pointing this out, and emphasis on small test file :)

Reply:
So all my documentation essentially ignores the option to zip any of it, for this reason. But good point to specifically point it out and make note that these files should be small.